File: app/src/main/java/com/returnnotfound/stain/di/module/FragmentBuilder.java
package com.returnnotfound.stain.di.module;

import com.returnnotfound.stain.presentation.main.MainFragment;
import com.returnnotfound.stain.presentation.main.discover.DiscoverFragment;
import com.returnnotfound.stain.presentation.main.love.LoveFragment;
import com.returnnotfound.stain.presentation.main.rank.RankFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class FragmentBuilder {

  @ContributesAndroidInjector
  abstract MainFragment contributeMainFragment();

  @ContributesAndroidInjector
  abstract DiscoverFragment contributesDiscoverFragment();

  @ContributesAndroidInjector
  abstract RankFragment contributesRankFragment();

  @ContributesAndroidInjector
  abstract LoveFragment contributesLoveFragment();
}

File: app/src/main/java/com/returnnotfound/stain/di/module/FragmentBuilder.kt
package com.returnnotfound.stain.di.module

import com.returnnotfound.stain.presentation.ui.main.MainFragment
import com.returnnotfound.stain.presentation.ui.main.discover.DiscoverFragment
import com.returnnotfound.stain.presentation.ui.main.love.LoveFragment
import com.returnnotfound.stain.presentation.ui.main.rank.RankFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class FragmentBuilder {
  @ContributesAndroidInjector
  abstract fun contributeMainFragment(): MainFragment

  @ContributesAndroidInjector
  abstract fun contributesDiscoverFragment(): DiscoverFragment

  @ContributesAndroidInjector
  abstract fun contributesRankFragment(): RankFragment

  @ContributesAndroidInjector
  abstract fun contributesLoveFragment(): LoveFragment
}